### PR TITLE
Chore/refactor away unstated to fix SSR

### PR DESCRIPTION
**Pull request by ryami333**
First of all, apologies for the monstrous PR and thanks in advance for your review(s)!

This PR refactors away any third-party state management and context management (the incumbent was `unstated`). The fundamental reasoning behind the the change was that `unstated` changed to an [asynchronous model](https://github.com/jamiebuilds/unstated/blob/cb67dbf454606b78eb31e548ddcc4f1fe0dc4a59/src/unstated.js#L21) during one of their releases which broke server-side rendering (see https://github.com/springload/react-accessible-accordion/issues/123 for example) and also forced us to do some pretty [hacky stuff](https://github.com/springload/react-accessible-accordion/blob/master/src/unstated-test-helpers.js) in our testing suite to make time-based assertions. In earlier releases `unstated` showed real promise, but (on top of the async issues) the maintainer refuses to enable issues or document releases/changes anywhere so it was a risky dependency to continue to maintain, long term.

The following were givens:
- We still need 'context' because an `AccordionItem` needs to know which `Accordion` it belongs to, and in order for 'true' accordions to work (only one item may be open at one time).
- We want to continue to support React 15 for a little longer, so we can't use the new shiny React 16.3+Context API. I plan to deprecate support for R15, but I would still like to maintain 16.0-16.2 support for a little while longer anyway.

So, I went with the legacy context API, which according to documentation _will_ be supported for the remainder of React 16 releases. I used a little workaround to bypass `prop-types` support, because we use Flow in the internals anyway, and because I wanted to keep the library's dependency footprint small.

I imagine that it will be very difficult to review the unit tests by viewing the diffs, as the changes are so significant that the negative diffs are unlikely to give any helpful context (mostly because they're so `unstated`-specific).

Don't hesitate to get in touch if you have any more questions about this PR :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,8 +111,7 @@
   },
   "dependencies": {
     "classnames": "^2.2.5",
-    "consecutive": "^5.0.4",
-    "unstated": "^2.1.1"
```
Comment (ryami333) [inline package.json]: 🎉

**Comment by ryami333** [comment]
Gidday @madarauchicha, basically it's just awaiting review from one of my colleagues, who has indicated that they will check it out sometime this week. I would welcome more reviews from anyone though, so if you've got a bit of time then any feedback would be appreciated 🙂

File: src/ItemContainer/ItemContainer.js [hunk 2]
```diff
@@ -1,32 +1,54 @@
...
 
-let nextUuid = consecutive();
-export function resetNextUuid() {
-    nextUuid = consecutive();
-}
+export type ItemContainer = {
+    uuid: UUID,
+};
+
+// Arbitrary, but ought to be unique to avoid context namespace clashes.
```
Comment (holloway) [inline src/ItemContainer/ItemContainer.js]: :+1:

File: src/AccordionItemBody/AccordionItemBody.wrapper.js [hunk 3]
```diff
@@ -20,29 +28,19 @@ class AccordionItemBodyWrapper extends Component<
...
     render() {
-        return (
-            <Subscribe to={[AccordionContainer, ItemContainer]}>
-                {this.renderItemBody}
-            </Subscribe>
-        );
+        const { itemStore, accordionStore, ...rest } = this.props;
+        const { uuid } = itemStore;
+        const { items, accordion } = accordionStore;
+        const item = items.filter(stateItem => stateItem.uuid === uuid)[0];
```
Comment (holloway) [inline src/AccordionItemBody/AccordionItemBody.wrapper.js]: so not `.find` because that would require a polyfill?

Comment (ryami333) [reply src/AccordionItemBody/AccordionItemBody.wrapper.js]: Exactly - no IE support for Array.prototype.find.

File: src/AccordionItem/AccordionItem.js [hunk 4]
```diff
@@ -48,7 +48,8 @@ class AccordionItem extends Component<AccordionItemProps, *> {
             ...rest
         } = this.props;
 
-        const currentItem = accordionStore.state.items.filter(
+        // Deliberately not using 'find' because IE compat.
```
Comment (holloway) [inline src/AccordionItem/AccordionItem.js]: Ah ignore this comment then https://github.com/springload/react-accessible-accordion/pull/128/files#r225007869

File: src/AccordionItem/AccordionItem.js [hunk 5]
```diff
@@ -14,8 +14,8 @@ type AccordionItemProps = ElementProps<'div'> & {
     accordionStore: AccordionContainer,
 };
 
-class AccordionItem extends Component<AccordionItemProps, *> {
-    componentDidMount() {
+class AccordionItem extends Component<AccordionItemProps> {
+    componentWillMount() {
```
Comment (holloway) [inline src/AccordionItem/AccordionItem.js]: Why the change from didMount to willMount?

File: src/AccordionContainer/AccordionContainer.js [hunk 6]
```diff
@@ -78,34 +95,50 @@ class AccordionContainer extends Container<StoreState> {
         }));
 
     setExpanded = (key: string | number, expanded: boolean) =>
```
Comment (holloway) [inline src/AccordionContainer/AccordionContainer.js]: use `UUID` flow type?

File: src/AccordionContainer/AccordionContainer.js [hunk 7]
```diff
@@ -1,46 +1,63 @@
...
+    items?: Array<Item>,
+};
+
+export type AccordionContainer = {
     accordion: boolean,
-    onChange: Function,
+    items: Array<Item>,
+    addItem: Item => void,
+    removeItem: (string | number) => void,
+    setExpanded: (key: string | number, expanded: boolean) => void,
```
Comment (holloway) [inline src/AccordionContainer/AccordionContainer.js]: `UUID` type again?

File: src/AccordionItem/AccordionItem.js (hunk 5 above)
Comment (ryami333) [reply src/AccordionItem/AccordionItem.js]: Rebased away!

File: src/AccordionContainer/AccordionContainer.js (hunk 6 above)
Comment (ryami333) [reply src/AccordionContainer/AccordionContainer.js]: Replaced everywhere!
